### ISSUE TITLE
Package suddenly stopped working; Maximum execution time of 2 seconds exceeded

### ISSUE THREAD

**Issue by olssonm** [spatie/laravel-backup#457]
Been using the laravel backup-package on multiple sites for the past year or so, but the other day one of apps using it started to acting up, whenever i run the `php artisan backup:run` command I got the following error:

```
[Symfony\Component\Debug\Exception\FatalErrorException]
Maximum execution time of 2 seconds exceeded
```

Very odd that this suddenly popped up, as I have no 2-second limits anywhere on the server or application (CLI is set to "0" and FPM is at 30s). 2-seconds is kind of drastic too – no? Anyhow, this is was on the production server. 

So took down the latest backup and tried from the development server and also in Homestead – again, same issue. 

Here is the stack trace logged:

```
PHP Warning:  Unknown: Cannot destroy the zip context in Unknown on line 0
PHP Stack trace:
PHP   1. {main}() /home/vagrant/SITE/artisan:0
PHP   2. Illuminate\Foundation\Console\Kernel->handle() /home/vagrant/SITE/artisan:36
PHP   3. Symfony\Component\Console\Application->run() /home/vagrant/SITE/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php:121
PHP   4. Symfony\Component\Console\Application->doRun() /home/vagrant/SITE/vendor/symfony/console/Application.php:116
PHP   5. Symfony\Component\Console\Application->doRunCommand() /home/vagrant/SITE/vendor/symfony/console/Application.php:185
PHP   6. Spatie\Backup\Commands\BaseCommand->run() /home/vagrant/SITE/vendor/symfony/console/Application.php:817
PHP   7. Illuminate\Console\Command->run() /home/vagrant/SITE/vendor/spatie/laravel-backup/src/Commands/BaseCommand.php:22
PHP   8. Symfony\Component\Console\Command\Command->run() /home/vagrant/SITE/vendor/laravel/framework/src/Illuminate/Console/Command.php:155
PHP   9. Illuminate\Console\Command->execute() /home/vagrant/SITE/vendor/symfony/console/Command/Command.php:261
PHP  10. Illuminate\Container\Container->call() /home/vagrant/SITE/vendor/laravel/framework/src/Illuminate/Console/Command.php:169
PHP  11. call_user_func_array:{/home/vagrant/SITE/vendor/laravel/framework/src/Illuminate/Container/Container.php:508}() /home/vagrant/SITE/vendor/laravel/framework/src/Illuminate/Container/Container.php:508
PHP  12. Spatie\Backup\Commands\BackupCommand->handle() /home/vagrant/SITE/vendor/laravel/framework/src/Illuminate/Container/Container.php:508
PHP  13. Spatie\Backup\Tasks\Backup\BackupJob->run() /home/vagrant/SITE/vendor/spatie/laravel-backup/src/Commands/BackupCommand.php:47
PHP  14. Spatie\Backup\Tasks\Backup\BackupJob->createZipContainingAllFilesToBeBackedUp() /home/vagrant/SITE/vendor/spatie/laravel-backup/src/Tasks/Backup/BackupJob.php:149
PHP  15. Spatie\Backup\Tasks\Backup\BackupJob->addDatabaseDumpsToZip() /home/vagrant/SITE/vendor/spatie/laravel-backup/src/Tasks/Backup/BackupJob.php:168
PHP  16. Illuminate\Support\Collection->each() /home/vagrant/SITE/vendor/spatie/laravel-backup/src/Tasks/Backup/BackupJob.php:202
PHP  17. Spatie\Backup\Tasks\Backup\BackupJob->Spatie\Backup\Tasks\Backup\{closure}() /home/vagrant/SITE/vendor/laravel/framework/src/Illuminate/Support/Collection.php:228
PHP  18. Spatie\Backup\Tasks\Backup\Zip->add() /home/vagrant/SITE/vendor/spatie/laravel-backup/src/Tasks/Backup/BackupJob.php:201
PHP  19. Spatie\Backup\Tasks\Backup\Zip->close() /home/vagrant/SITE/vendor/spatie/laravel-backup/src/Tasks/Backup/Zip.php:94
PHP  20. ZipArchive->close() /home/vagrant/SITE/vendor/spatie/laravel-backup/src/Tasks/Backup/Zip.php:68
```

The live server is on Laravel Forge, running 7.0.12. Same for the development server. The trace above is from Laravel Homestead (old version running 5.6.7 which somehow encountered the exact same issue). 

The database is MySQL and rather large (~175MB) and the backups are stored on AWS S3. 

Is there something I'm missing? How can this so suddenly happen? Any ideas on how I can debug this or where to start looking?

☹️

**Comment by freekmurze** [comment]
This might be related to that big database you're trying to back up.

Could you try adding a dump timeout to the configuration of your database? By default it should be `60` but it's worth giving it a shot.

The documentation has an example on [how to manually set the timeout of a db dumper](https://docs.spatie.be/laravel-backup/v4/installation-and-setup#dumping-the-database)

If that does not help, could you let me know which version of the package you are running?

**Comment by olssonm** [comment]
Thanks for the reply – however it didn't work (I ran 3.10.2 previously, updated to 3.11.0 when I detected the issue, albeit with no luck).

What I did was just to delete the entire installation locally at first, then clone the repo and re-install all the dependancies. And things just worked...? =/ So I'm finalizing the local testing here before I do the same on the live server. 

So, this issue might not belong to the spatie/laravel-backup-package at all, just that it was there I detected it. How very odd.

Thanks for your feedback, and sorry for not testing this first perhaps =)
